uiowa/uiowa
# Dispatch/Facilities Message Logging

Resolves #6855

<img width="951" alt="Screenshot 2023-09-20 at 11 20 52 AM" src="https://github.com/uiowa/uiowa/assets/4663676/2c3839c0-cff8-497a-bac7-7f5859687f3c">


# To Test

```
ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local uli admin/config/sitenow/sitenow-dispatch
```
Go to [facilities.uiowa.edu](https://www.facilities.uiowa.edu/admin/config/services/facilities-alerts-emails) and get the dispatch API key and enter it here https://facilities.uiowa.ddev.site/admin/config/sitenow/sitenow-dispatch

Create two alerts. Attempt to schedule one of them through the notifications tab.

Note that the unschedule alert does not display a log but the schedule alert does. See that the message id is linked to the modal within the dispatch communication. Also note that the user id gets output as the user name.

Inspect the database table: `facilities_uiowa_edu.sitenow_dispatch_messages_log`

Schedule another dispatch message for the same alert. See that the table sort works as expected and that the tab count is incremented.
Schedule the previously unscheduled alert.

Delete the alert with two scheduled messages and inspect the database table again. The two entries for that entity id should be removed leaving the other log entry for the other alert.

Visit the dispatch interface and unschedule each of the messages you scheduled.
